src/caffeine/prif_queries_s.f90

```diff
@@ -13,6 +18,16 @@
   end procedure
 
   module procedure prif_base_pointer
+    integer(c_int) :: num_img
+
+    call prif_num_images(image_count=num_img)
+    call assert(image_num .gt. 0 .or. image_num .lt. num_img, "prif_base_pointer: image_num not within valid range")
```

Comment:
I believe there are two off-by-one mistakes in the assertion expression above.

Please add tests to detect these defects before fixing them.

Reply:
To be more specific, I think the "gets null for image 0" covers the first one, and changing the "gets valid pointer" to use `num_img` covers the second.

Reply:
Yes, that assert was wrong for 3 reasons. I have updated it and added tests as noted, I believe the logic is correct now, but of course this should be checked by others as well.